### ISSUE TITLE
[2.3.3-0] enabling both "show bold as font" and "show bold as colour" does not work

### ISSUE THREAD

**Issue by wmwdld** [mintty/mintty#536]
I have both of the above options selected.  In previous versions, this worked: bold text was rendered both in a lighter color and with a heavier typeface.  In 2.3.3 it honors only the color, not the bold.  If I disable "show bold as color", then it successfully uses the bold version of the font.  But I don't want that, I want both like in the previous versions.

The font I'm using is Inconsolata 12 pt medium.

(edit: I made a typo in the version, now fixed)

**Comment by mintty** [comment]
Please check your description; with both options, I see it honour the bold but not the colour.
This is documented behaviour, see the manual page about options `BoldAsColour` and `BoldColour`; do you have a value assigned to `BoldColour`?
There had been some changes in handling the bold attribute, but not very recently. They were settled with 2.2.1 (Nov 2015).
Also, note that both bold font *and* colour are applied for characters with an explicit ANSI colour attribute (as documented).

**Comment by wmwdld** [comment]
I'm sorry, I don't understand your response.  All I know was that mintty was working fine, and then I upgraded and it broke.  I didn't change a thing in the config.  Here is my config file:

    BoldAsFont=yes
    Transparency=off
    OpaqueWhenFocused=no
    CursorType=block
    Font=Inconsolata
    FontHeight=12
    FontSmoothing=default
    Columns=160
    Rows=54
    CursorBlinks=no
    Locale=en_US
    Charset=UTF-8
    Term=xterm-256color
    ForegroundColour=255,255,255
    BackgroundColour=48,10,36
    Black=0,0,0
    Red=204,0,0
    Green=78,154,6
    Yellow=196,160,0
    Blue=52,101,164
    Magenta=117,80,123
    Cyan=6,152,154
    White=211,215,207
    BoldBlack=64,64,64
    BoldRed=239,41,41
    BoldGreen=138,226,52
    BoldYellow=252,233,79
    BoldBlue=114,159,207
    BoldMagenta=173,127,168
    BoldCyan=52,226,226
    BoldWhite=238,238,236
    BoldAsColour=yes
    CursorColour=0,170,0
    BellTaskbar=no
    FontWeight=500
    FontIsBold=no

And I'm referring to text that has an explicit color and bold attribute, such as executable files in the output of `ls` (e.g. `\e[01;32m`)

**Comment by mintty** [comment]
This happens with fonts that provide only one weight. It works as it used to with fonts that have two weights. Obviously, normal (uncoloured) bold still uses the Windows-boldened font version, so the coloured bold should do that too.

**Comment by wmwdld** [comment]
So what you're saying is that despite the font selection dialog presenting four styles for Inconsolata (medium, medium oblique, bold, bold oblique), only one of them is actually provided by the font and the rest are faked/synthesized by Windows?  And mintty isn't currently considering the faked variants when rendering text colored by ANSI sequences?  Have I got that right?

Update: before posting I checked Google Fonts and there's a newer version of Inconsolata than what I have installed and it has a true bold variant.  I installed that and now mintty works like it used to and shows bold again.  (But everything is a bit different, I guess because the newer variant has tweaked metrics, and because of the no longer fake bold.  It'll take some time to get used to this.)

Anyway, thanks.
